pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -87,13 +89,14 @@ func (a *index) serviceServiceBuilder(
 		waypointStatus.Error = wperr
 
 		a.networkUpdateTrigger.MarkDependant(ctx) // Mark we depend on out of band a.Network
-		return &model.ServiceInfo{
-			Service:       a.constructService(s, waypoint),
+		svc := a.constructService(s, waypoint)
+		return precomputeServicePtr(&model.ServiceInfo{
```

Comment:
Would be nice to have control plane CPU perf tests as part of our long-running testing process. Want to try and avoid a scenario where a refactor occurs and we no longer precompute the service

Reply:
Maybe a quick version is just a unit tests that calls these builder functions and fails if `MarshaledAddress` and `AsAddress` are nil

Reply:
Working on the benchmarks. TBD where I get, else I will fallback to the unit test. Or perhaps both

Reply:
https://github.com/istio/istio/pull/54549. ill add a test too